pkg/index/job/correction/service/corrector.go

```diff
@@ -181,15 +186,16 @@ func (c *correct) correct(ctx context.Context) (err error) {
 			bconcurrency := c.cfg.Corrector.GetBboltAsyncWriteConcurrency()
 			bolteg.SetLimit(bconcurrency)
 
-			var mu sync.Mutex
 			log.Infof("starting correction for agent %s, stream concurrency: %d, bbolt concurrency: %d", addr, sconcurrency, bconcurrency)
 
 			vc := vald.NewValdClient(conn)
 			stream, err := vc.StreamListObject(ctx, &payload.Object_List_Request{})
 			if err != nil {
+				jobErrs = append(jobErrs, err)
```

Comment:
Using `named return` and the `defer`,  add to `jobErrs` when an error is not nil at the time the closure function exits, I think we can prevent unexpected bugs!!